templates/Audio/ID3v2.tcl

```diff
@@ -0,0 +1,672 @@
+# ID3v2 binary template
+#
+# Specification can be found at:
+# http://id3.org/Developer%20Information
+#
+# Copyright (c) 2019 Mattias Wadman
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of
+# this software and associated documentation files (the "Software"), to deal in
+# the Software without restriction, including without limitation the rights to
+# use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+# of the Software, and to permit persons to whom the Software is furnished to do
+# so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+# TODO: footer?
+# TODO: unsynchronise flag? messy todo
+# TODO: redo sections? "Frames" section etc?
+# TODO: flags, bit fields helper?
+
+proc syncsafeint32 { {name ""} } {
+    set n [uint32]
+    # syncsafe integer is a number encoded
+    # with 8th bit in each byte set to zero
+    # 0aaaaaaa0bbbbbbb0ccccccc0ddddddd ->
+    # 0000aaaaaaabbbbbbbcccccccddddddd
+    set v [expr \
+        (($n & 0x7f000000) >> 3) | \
+        (($n & 0x007f0000) >> 2) | \
+        (($n & 0x00007f00) >> 1) | \
+        (($n & 0x0000007f) >> 0) \
+    ]
+    if { $name != "" } {
+	    entry $name [format "%d (%d)" $v $n] 4 [expr [pos]-4]
+    }
+    return $v
+}
+
+proc uint24 { {name ""} } {
```

Comment:
Should be a native command?

Reply:
`uint24` is already built-in, but I guess I forgot to document it.

Reply:
Aha also didn't noticed. Fixed now.